### PR TITLE
PEK 764 cleanup and get snap working

### PR DESCRIPTION
**Pull request by spmcmillan**
* Move pytests to tests directory
* Get snap/checkbox to properly find the installed snap tests with
  checkbox
* Move pytests to use tox instead of our custom run.sh script
* Add pytest categories to call via "tox -e <category>"
* Add checkbox categories to call via
  "checkbox-tdx-classic.test-runner-automated-<category>"
* Get rid of redundant TDXTEST_IMAGE_FILE environment variable and use
  existing TDXTEST_GUEST_IMG (and don't copy it)
* Use relative checkbox-cli path to be compatible to using other runtime
* In pytest, check for image specification and existence
* Make sure TDXTEST_GUEST_IMG variable is passed to tox
  environments (e.g. checkbox22, checkbox24, ...)

### CODE REVIEW

**Review by hector-cao — COMMENTED** [review]
Since I m off for some days, i will let @frenchwr review it,

File: tests/checkbox/checkbox-provider-tdx/units/tests/jobs.pxu [hunk 1]
```diff
@@ -83,20 +83,20 @@ command:
...
-flags: simple
-_summary: Test boot stress
-depends:
-after:
-requires:
-  executable.name == 'qemu-system-x86_64'
-command:
-  export LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu/:$LD_LIBRARY_PATH
-  setup-env-and-run test_stress_boot.py
+#id: tdx-stress/td-stress-boot
```
Comment (frenchwr) [inline tests/checkbox/checkbox-provider-tdx/units/tests/jobs.pxu:86]: Did you temporarily comment this out and forget to remove?

**Review by frenchwr — COMMENTED** [review]
Looks awesome! 

I haven't actually tested the code, though, which I'd like to do. Where would you suggest I do that? Maybe right-glider? If so would you mind adding my keys?

File: tests/tox.ini [hunk 2]
```diff
@@ -0,0 +1,52 @@
+[tox]
+envlist = 
+  test_host
+isolated_build = True
+skip_missing_interpreters = True
+
+[testenv]
+allowlist_externals=bash
+pass_env = TDXTEST_GUEST_IMG
+deps =
```
[the remark this answers is not in the document]
Comment (spmcmillan) [reply tests/tox.ini:10]: I pinned them now.  I think that makes sense.  I battle this kind of thing in my head.  Restrictive versus letting it upgrade and "hope" it works.

File: tests/tox.ini [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+commands = 
+  python3 -m pytest -s -v --junitxml=test_stress_report.xml -k test_stress
+
+[testenv:test_all]
+commands = 
+  python3 -m pytest -s -v --junitxml=test_all_report.xml
+
+[testenv:test]
+commands = 
+  python3 -m pytest -s -v --junitxml=test_quote_report.xml -k {posargs}
```
[the remark this answers is not in the document]
Comment (spmcmillan) [reply tests/tox.ini]: Gives the flexibility to specify exactly the desired tests using pytest logic.  So, if I want to run guest and host tests only I could do it through this line.  Or I can use regular expressions here too.  Gives flexibility for those comfortable with tox and pytest.  I did change this to test_specify and fixed the junit xml report name

File: tests/checkbox/checkbox-provider-tdx/units/tests/jobs.pxu (hunk 1 above)
Comment (spmcmillan) [reply tests/checkbox/checkbox-provider-tdx/units/tests/jobs.pxu:86]: The td-stress-boot takes a VERY long to complete.  So I commented it out for now and was planning on going back to it and maybe better categorizing it.  Some tests fail too so there is more work to do.  This is an attempt to get the framework.

File: tests/tests/setup-env-and-run [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+TMP_FOLDER=/tmp/snap/checkbox-tdx-classic
+SCRIPT_DIR=$SNAP/providers/checkbox-provider-tdx/bin/
+
+setup_venv() {
+  mkdir -p ${TMP_FOLDER}/.pytest_cache
+  mkdir -p ${TMP_FOLDER}/reports
+  chmod a+w ${TMP_FOLDER}/reports
+  python3 -m venv ${TMP_FOLDER}/venv
+  source ${TMP_FOLDER}/venv/bin/activate
+  python3 -m pip install paramiko==3.3.1 \
```
[the remark this answers is not in the document]
Comment (spmcmillan) [reply tests/tests/setup-env-and-run:36]: this is just for checkbox side

File: tests/tests/pytest_prep.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    if [ -f "$f/qemu.pid" ]; then
+      pkill -TERM -F "$f/qemu.pid"
+    fi
+    rm -rf $f
+  done
+  rm -rf /tmp/tdxtest-*
+  rm -rf /run/user/$UID/tdxtest-*
+}
+
+install_deps() {
```
[the remark this answers is not in the document]
Comment (spmcmillan) [reply tests/tests/pytest_prep.sh]: Those "apt" installations are not done in tox.ini so they are valid in this script.

File: tests/README.md [hunk 6]
```diff
@@ -20,32 +25,100 @@ Go to the `tests` folder.
 - Run sanity tests to check the host setup:
 
 ```
-$ sudo ./run.sh pytest pytest/test_host_*.py
+$ sudo tox -e test_host
```
Comment (frenchwr) [inline tests/README.md:28]: I needed to pass -E to sudo to get this to work, with `TDXTEST_GUEST_IMG` exported to my environment. I think this should be added to the README for consistency with the checkbox section.

**Review by frenchwr — COMMENTED** [review]
One small comment for the README. Only other thing I would advocate for...right now each tox category (test_host, test_boot, etc) is creating its own virtualenv in the `tests/.tox/` directory even through each virtualenv is identical. I think there is a way to tell tox to share the same virtualenv among all the test categories. That will save us some disk space and time.

My tests are still running on right-glider but I don't think any failures would be related to this PR.

File: tests/README.md (hunk 6 above)
Comment (spmcmillan) [reply tests/README.md:28]: Good idea, just added this

**Review by frenchwr — APPROVED** [review]
LGTM!

**Pull request merged**